### PR TITLE
Add Rollbar, filter sensitive params

### PR DESCRIPTION
**Pull request by mathias**
This seemed useful on my fork, and might as well filter the params in Rails logs, too.

### CODE REVIEW

File: config/initializers/rollbar.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    '_session',
+    'access_token',
+    'oauth_grant.code',
+    'refresh_token',
+    'request.session',
+    'request.session.csrf',
+    'request.session.resource.values.access_token_secret',
+    'request.session.resource.values.oauth_grant_secret',
+    'request.session.resource.values.refresh_token_secret',
+    'request.session.resource.values.slow_db_password',
```
Comment (jessieay) [inline config/initializers/rollbar.rb]: this doesn't seem relevant?

File: config/initializers/filter_parameter_logging.rb [hunk 2]
```diff
@@ -1,4 +1,21 @@
...
+  '_session',
+  'access_token',
+  'oauth_grant.code',
+  'refresh_token',
+  'request.session',
+  'request.session.csrf',
+  'request.session.resource.values.access_token_secret',
+  'request.session.resource.values.oauth_grant_secret',
+  'request.session.resource.values.refresh_token_secret',
+  'request.session.resource.values.slow_db_password',
```
Comment (jessieay) [inline config/initializers/filter_parameter_logging.rb]: not needed here, is it?

**Comment by mathias** [comment]
@jessieay Whoops, those are from copy-paste without thinking.

File: config/initializers/rollbar.rb (hunk 1 above)
Comment (mathias) [reply config/initializers/rollbar.rb]: Fixed in e244e9168ecd7555781df0b9c50c0ad69e745f0c. Copy-paste error!

File: config/initializers/rollbar.rb [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    :password,
+    :session
+  ]
+  config.scrub_headers |= [
+    'Authorization',
+    'Cookies',
+    'HTTP_AUTHORIZATION',
+    'HTTP_X_CSRF_TOKEN',
+    'X_CSRF_TOKEN',
+    'omniauth.auth',
```
Comment (jessieay) [inline config/initializers/rollbar.rb]: we aren't using omniauth here, are we?

File: config/initializers/rollbar.rb [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  config.access_token = ENV['ROLLBAR_ACCESS_TOKEN']
+
+  # Here we'll disable in 'test' and dev:
+  if !Rails.env.production?
+    config.enabled = false
+  end
+
+  config.environment = ENV["APP_ENVIRONMENT"] || "development"
+
+  # By default, Rollbar will try to call the `current_user` controller method
```
Comment (jessieay) [inline config/initializers/rollbar.rb]: do you remind removing all of these comments? I find it easier to read these config files if they are free of documentation comments

File: config/initializers/rollbar.rb [hunk 5]
```diff
@@ -0,0 +1,85 @@
+require 'rollbar/rails'
+
+Rollbar.configure do |config|
+  # Without configuration, Rollbar is enabled in all environments.
+  # To disable in specific environments, set config.enabled=false.
+
+  config.access_token = ENV['ROLLBAR_ACCESS_TOKEN']
+
+  # Here we'll disable in 'test' and dev:
+  if !Rails.env.production?
```
Comment (jessieay) [inline config/initializers/rollbar.rb]: thoughts on enabling it based on an env var instead? I am imagining a scenario in which someone forks this but is not using rollbar in prod.

File: config/initializers/rollbar.rb (hunk 3 above)
Comment (mathias) [reply config/initializers/rollbar.rb]: Good catch. I'm pulling in fields from the internal gem, which we can't use here, and it is a "blanket" for a reason -- it just tries to cover as many possible leaks as possible.

My fork will probably continue to use this, as I need to demo a new dashboard behind a Herokai-only auth check.

I'll remove it here, though.

File: config/initializers/rollbar.rb (hunk 4 above)
Comment (mathias) [reply config/initializers/rollbar.rb]: Yep. These came from another add-on service that I suspect used the templated Rollbar config file.

**Comment by mathias** [comment]
Comments above should all be addressed or replied to, @jessieay .

File: config/initializers/filter_parameter_logging.rb [hunk 6]
```diff
@@ -1,4 +1,13 @@
 # Be sure to restart your server when you modify this file.
 
 # Configure sensitive parameters which will be filtered from the log file.
-Rails.application.config.filter_parameters += [:password]
+Rails.application.config.filter_parameters += [
+  '_session_id',
```
Comment (mathias) [inline config/initializers/filter_parameter_logging.rb]: `_session_id` is what sudo sandwich is using when one runs `heroku addons:open sudo-sandwich` so I wanted to be sure it never makes it to third party Rollbar here.

File: config/initializers/rollbar.rb [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  config.environment = ENV["APP_ENVIRONMENT"] || "development"
+
+  config.exception_level_filters.merge!({
+    'ActionController::RoutingError' => 'ignore',
+    'ActionController::UnknownHttpMethod' => 'ignore',
+    'ActionController::BadRequest' => 'ignore',
+    'ActionView::MissingTemplate' => 'ignore'
+  })
+
+  config.scrub_fields |= [
```
[the remark this answers is not in the document]
Comment (mathias) [reply config/initializers/rollbar.rb]: Yeah -- that's what the blanket gem gets us, but it is not open source. My worry here is depending on alphabetical file load order, which I've never liked about Rails initializers. (One way to ensure that load order is right is to set the constant in `filter_parameter_logging.rb` and rename `rollbar.rb` to `z_rollbar.rb`. But I'm not a huge fan of that, either.

File: config/initializers/rollbar.rb (hunk 5 above)
Comment (mathias) [reply config/initializers/rollbar.rb]: Yes, I've just smoke tested it, and the rollbar agent doesn't complain in the regular log if it has no config var. When an exception occurs, it does add to the logs:
```
[Rollbar] Got unexpected status code from Rollbar api: 400
[Rollbar] Response: {
  "err": 1,
  "message": "access token required"
}
```

**Pull request merged**